dev/io.openliberty.wsoc.2.1.internal_fat/test-applications/basic21.war/src/basic/war/UpgradeServlet.java

```diff
@@ -12,34 +12,30 @@
 package basic.war;
 
 import java.io.IOException;
-import java.util.Collections;
 import java.util.HashMap;
-import java.util.Map;
 
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import jakarta.websocket.DeploymentException;
 import jakarta.websocket.server.ServerContainer;
 import jakarta.websocket.server.ServerEndpointConfig;
 
-import jakarta.websocket.DeploymentException;
-
 /**
  * Upgrade to ws using the Websocket 2.1 ServerContainer#upgradeHttpToWebSocket method
  */
 public class UpgradeServlet extends HttpServlet {
 
     @Override
-    public void doGet(HttpServletRequest req, HttpServletResponse resp)
-                    throws ServletException, IOException {
+    public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 
         ServerContainer container = (ServerContainer) req.getServletContext().getAttribute("jakarta.websocket.server.ServerContainer");
-        try{
-            // echo endpoint is used into another test, so I am reusing it here. 
+        try {
+            // echo endpoint is used into another test, so I am reusing it here.
             container.upgradeHttpToWebSocket(req, resp, ServerEndpointConfig.Builder.create(EchoServerEP.class, "/echo").build(), new HashMap<String, String>());
-        } catch(DeploymentException ex){
-            
+        } catch (DeploymentException ex) {
+            req.getServletContext().log(ex.toString());
```

Comment:
I haven't seen this log call before. That's convenient. I see it just logs here the message here then: https://github.com/OpenLiberty/open-liberty/blob/6a5454d104f304cbae127abce326a8d9d049fbcd/dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java#L3442 

We don't need any to make any message here right? (or any translations)?

Reply:
We're just logging a message from a test Servlet so need for any sort of translation. Does that make sense @volosied ?